### PR TITLE
feature/PP-45

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Solve circular dependency found when including unit tests in rif-relay-contracts
- Add unit testing for each dependency for the Rif Relay Project

## Why

- Circular dependencies must be avoided in most of the situations
- Each project needs to have their own way of testing

## Refs

- [Jira issue 45](https://rsklabs.atlassian.net/browse/PP-45)
- [Jira issue 236](https://rsklabs.atlassian.net/browse/PP-236)
- https://github.com/rsksmart/rif-relay-contracts/pull/43
- https://github.com/rsksmart/rif-relay-common/pull/24

### CODE REVIEW

File: src/Configurator.ts [hunk 1]
```diff
@@ -53,6 +53,9 @@ const defaultEnvelopingConfig: EnvelopingConfig = {
 export function configure(
     partialConfig: Partial<EnvelopingConfig>
 ): EnvelopingConfig {
+    console.log('-------------------------------');
+    console.log(constants);
+    console.log('-------------------------------');
```
Comment (antomor) [inline src/Configurator.ts:58]: Could we remove those log lines?

File: src/RelayClient.ts [hunk 2]
```diff
@@ -291,6 +288,7 @@ export class RelayClient {
         let deployCallEstimate = 0;
 
         if (isSmartWalletDeploy) {
+            console.log('si llegue aca bien y sin novedad2');
```
Comment (antomor) [inline src/RelayClient.ts:291]: I think this line could be removed

File: src/RelayClient.ts [hunk 3]
```diff
@@ -304,6 +302,7 @@ export class RelayClient {
                 trxDetails.tokenGas
             );
         } else {
+            console.log('si llegue aca bien y sin novedad3');
```
Comment (antomor) [inline src/RelayClient.ts:305]: I think this line could be removed

File: src/RelayClient.ts [hunk 4]
```diff
@@ -386,7 +385,7 @@ export class RelayClient {
         ) {
             throw new Error('Request type is not for SmartWallet deploy');
         }
-
+        console.log(' si estamos pasando x aca ');
```
Comment (antomor) [inline src/RelayClient.ts:388]: I think this line could be removed

**Review by antomor — CHANGES_REQUESTED** [review]
Although the tests are running properly, I think we have some build issues. Could you please review them?

File: src/Configurator.ts (hunk 1 above)
Comment (franciscotobar) [reply src/Configurator.ts:58]: sure, Ill remove it.

File: package.json [hunk 5]
```diff
@@ -27,8 +27,8 @@
     ]
   },
   "dependencies": {
-    "@rsksmart/rif-relay-contracts": "0.0.2-alpha.4",
-    "@rsksmart/rif-relay-common": "0.0.2-alpha.5",
+    "@rsksmart/rif-relay-contracts": "github:rsksmart/rif-relay-contracts#feature/PP-45",
+    "@rsksmart/rif-relay-common": "github:rsksmart/rif-relay-common#feature/PP-45",
```
Comment (antomor) [inline package.json:31]: we should update those lines before merging the PR

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**